src/resource.h

```diff
@@ -98,6 +98,11 @@ class Resource {
   /// Changes the product's package id
   virtual void ChangePackageId(int new_package_id = Package::unpackaged_id()) {};
 
+  /// Repackages a single resource into a package. If some quantity of the 
+  /// resource cannot be packaged using the given packaging strategy and
+  /// restrictions, the remainder is left in the resource object. 
+  std::vector<Resource::Ptr> Package(Package::Ptr pkg);
```

Comment:
This should be pure virtual:
```suggestion
  virtual std::vector<Resource::Ptr> Package(Package::Ptr pkg) = 0;
```